pkg/netpol/connlist/conns_formatter_txt.go

```diff
@@ -33,107 +40,144 @@ func (t *formatText) writeOutput(conns []Peer2PeerConnection, exposureConns []Ex
+	ingressExpLines := make([]string, 0)
+	egressExpLines := make([]string, 0)
 	unprotectedLines := make([]string, 0)
 	for _, ep := range sortedExposureResults {
 		// ingress and egress lines per peer, internally sorted
-		ingressLines, ingressUnprotectedLine := t.getPeerIngressExposureLines(ep)
-		exposureLines = append(exposureLines, ingressLines...)
-		if ingressUnprotectedLine != "" {
-			unprotectedLines = append(unprotectedLines, ingressUnprotectedLine)
-		}
-		egressLines, egressUnprotectedLine := t.getPeerEgressExposureLines(ep)
-		exposureLines = append(exposureLines, egressLines...)
-		if egressUnprotectedLine != "" {
-			unprotectedLines = append(unprotectedLines, egressUnprotectedLine)
-		}
-	}
-	res := "Exposure Analysis Result:\n"
-	res += strings.Join(exposureLines, newLineChar)
-	if len(unprotectedLines) > 0 {
-		res += "\n\nWorkloads which are not protected by network policies:\n"
-		sort.Strings(unprotectedLines)
-		res += strings.Join(unprotectedLines, newLineChar)
+		pIngressLines, pEgressLines, pUnprotectedLines := t.writePeerExposureAndUnprotectedLines(ep, maxPeerStrLen)
+		ingressExpLines = append(ingressExpLines, pIngressLines...)
+		egressExpLines = append(egressExpLines, pEgressLines...)
+		unprotectedLines = append(unprotectedLines, pUnprotectedLines...)
 	}
+	sort.Strings(unprotectedLines)
+	// results of exposure for all peers
+	res := exposureAnalysisHeader
+	res += writeExposureSubSection(egressExpLines, egressExpHeader)
+	res += writeExposureSubSection(ingressExpLines, ingressExpHeader)
+	res += writeExposureSubSection(unprotectedLines, unprotectedHeader)
 	return res
 }
 
-//nolint:dupl //same functionality but on different variables of exposedPeer and formatText so prefer to split
-func (t *formatText) getPeerIngressExposureLines(ep ExposedPeer) (ingressLines []string, ingressUnprotectedLine string) {
-	// if a peer is not protected, two lines are to be added to exposure analysis result:
-	// 1. all conns with entire cluster (added here)
-	// 2. all conns with ip-blocks (all destinations); for sure found in the ip conns map so will be added automatically
-	// also unprotected line will be added
-	if !ep.IsProtectedByIngressNetpols() {
-		ingressLines = append(ingressLines, formSingleExposureConn(ep.ExposedPeer().String(), entireCluster,
-			common.MakeConnectionSet(true), true).string())
-		ingressUnprotectedLine = ep.ExposedPeer().String() + " is not protected on Ingress"
-	} else { // protected
-		for _, data := range ep.IngressExposure() {
-			// for txt output append the string of the singleConnFields
-			ingressLines = append(ingressLines, formExposureItemAsSingleConnFiled(ep.ExposedPeer(), data, true).string())
-		}
-	}
-	// append ingress ip conns to this peer
-	if ipConns, ok := t.PeerToConnsFromIPs[ep.ExposedPeer().String()]; ok {
-		ingressLines = append(ingressLines, ipConns...)
+func (t *formatText) writePeerExposureAndUnprotectedLines(ep ExposedPeer, maxPeerStrLen int) (ingressLines,
```

Comment:
add description what this function returns

Reply:
done